plugins/scaffolder-backend/src/scaffolder/tasks/TaskWorker.ts

```diff
@@ -182,14 +190,44 @@ export class TaskWorker {
     });
   }
 
+  protected truncateParameters(parameters: JsonObject) {
+    const auditMaxLength =
+      this.config?.getOptionalNumber('scaffolder.auditor.maxLength') ?? 256;
+
+    if (auditMaxLength === -1) {
+      this.logger?.debug(
+        `scaffolder.auditor.maxLength manually disabled via configuration, no task parameter length limit set.`,
+      );
+      return parameters;
+    }
+
+    const truncatedParameters: JsonObject = {};
+
+    for (const key in parameters) {
+      if (Object.prototype.hasOwnProperty.call(parameters, key)) {
```

Comment:
`Object.hasOwn` is the nicer one to use nowadays

Reply:
done